src/components/Patient/PatientIndex.tsx

```diff
@@ -127,39 +127,57 @@ export default function PatientIndex({ facilityId }: { facilityId: string }) {
   const [identifierSearch, setIdentifierSearch] = useState<{
     config?: string;
     value?: string;
+    regex?: string;
   }>({});
 
-  const handleSearch = useCallback((key: string, value: string) => {
-    if (key === "phone_number") {
-      setPhoneNumberQuery({
-        phone_number: isValidPhoneNumber(value) || value === "" ? value : null,
-      });
-      setIdentifierSearch({});
-    } else {
-      setPhoneNumberQuery({ phone_number: "" });
-      setIdentifierSearch({ config: key, value });
+  const handleSearch = useCallback(
+    (key: string, value: string) => {
+      if (key === "phone_number") {
+        setPhoneNumberQuery({
+          phone_number:
+            isValidPhoneNumber(value) || value === "" ? value : null,
+        });
+        setIdentifierSearch({});
+      } else {
+        setPhoneNumberQuery({ phone_number: "" });
+        setIdentifierSearch({
+          config: key,
+          value,
+          regex: searchOptions.find((o) => o.key === key)?.regex,
+        });
+      }
+    },
+    [searchOptions],
+  );
+
+  const isIdentifierValid = (() => {
```

Comment:
🤔  We don't invoke anonymous arrow functions in our code base; try to follow our patterns 👍